contracts/helpers/order-helper/lib/OrderAvailabilityLib.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import {
+    ConsiderationInterface
+} from "seaport-types/src/interfaces/ConsiderationInterface.sol";
+
+import { AdvancedOrder } from "seaport-types/src/lib/ConsiderationStructs.sol";
+
+import { UnavailableReason } from "seaport-sol/src/SpaceEnums.sol";
+
+import { OrderStructureLib, State } from "./OrderStructureLib.sol";
+
+/**
+ * @notice Helper library for determining order availability.
+ */
+library OrderAvailabilityLib {
+    using OrderStructureLib for AdvancedOrder;
+
+    /**
+     * @notice Returns true if the order is available for fulfillment.
+     */
+    function isAvailable(
+        AdvancedOrder memory order,
+        ConsiderationInterface seaport
+    ) internal view returns (bool) {
+        return unavailableReason(order, seaport) == UnavailableReason.AVAILABLE;
+    }
+
+    /**
+     * @notice Returns the order's UnavailableReason. Available orders will
+     *         return UnavailableReason.AVAILABLE to indicate that they are
+     *         available for fulfillment.
+     */
+    function unavailableReason(
```

Comment:
don't we need to also account for an order that's unavailable due to a reverting generateOrder check? Maybe there's not a straightforward way to check for that?

Reply:
At the moment, this doesn't support contract orders at all—it just reverts early if any of the provided orders are contract orders.